LayoutKitExampleLayouts/CircleImagePileLayout.swift

```diff
@@ -36,13 +36,17 @@ public class CircleImagePileLayout: StackLayout {
                    viewReuseId: viewReuseId,
                    sublayouts: sublayouts)
     }
-
-    public override func makeView(from recycler: ViewRecycler, configure: Bool) -> UIView? {
-        switch mode {
-        case .leadingOnTop:
-            let view: CircleImagePileView = recycler.makeView(viewReuseId: viewReuseId)
-            return view
-        case .trailingOnTop:
+    
+    override public var needsView: Bool {
+        return mode == .leadingOnTop
+    }
+    
+    public override func makeView(from recycler: ViewRecycler) -> View? {
```

Comment:
hmm, I don't really like how there is duplicated logic here. Also, if a layout that does this type of thing is further subclassed, then there is no type safety if a subclass does a similar thing (can't enforce that subclass's type inherits from CircleImagePileView).

I am wondering if it would be better to do one (or both) of the following:

1. Discourage/prevent subclassing of Layouts (mark them as final or turn into structs). Then layouts like this would wrap StackLayout instead of inheriting it. Perhaps even get rid of BaseLayout at the cost of some property boilerplate.

2. Pass through the type parameter for all layouts (not just SizeLayout). e.g. `public class StackLayout<V: View>: BaseLayout<V>`. This example would customize the type as `CircleImagePileView` so that it is always used. Would need to adjust the implementation of `CircleImagePileView` to have a switch between the two behaviors. This is a bit annoying because it prevents nested types: e.g. `StackLayout.Distribution -> StackLayoutDistribution`.

Thoughts?